### PR TITLE
Add python-engineio_3 and python-socketio_4

### PR DESCRIPTION
**Pull request by graham33**
###### Motivation for this change
Socket.IO and Engine.IO protocols are not backwards-compatible and major versions of the client and server need to be coordinated.  It's therefore useful to have python-engineio 3.x and python-socketio 4.x available, but unstable has moved up to 4.x and 5.x respectively.  Happy to maintain these older versions.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-socketio/4.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  checkInputs = [
+    mock
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "socketio" ];
+
+  # pytestCheckHook seems to change the default log level to WARNING, but the
+  # tests assert it is ERROR
+  disabledTests = [ "test_logger" ];
```
Comment (graham33) [inline pkgs/development/python-modules/python-socketio/4.nix:35]: I couldn't see what was causing this (if I just use `pytest` rather than `pytestCheckHook` the tests pass), but it was easy enough to disable these tests.

File: pkgs/development/python-modules/python-socketio/default.nix [hunk 2]
```diff
@@ -36,7 +36,7 @@ buildPythonPackage rec {
       Socket.IO is a lightweight transport protocol that enables real-time
       bidirectional event-based communication between clients and a server.
     '';
-    homepage = "https://github.com/miguelgrinberg/python-engineio/";
+    homepage = "https://github.com/miguelgrinberg/python-socketio/";
```
Comment (graham33) [inline pkgs/development/python-modules/python-socketio/default.nix:39]: Noticed copy+paste error

**Comment by graham33** [comment]
@Mic92  FYI

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113247` at 6763803a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/1722e45b1dc5ec2866f0b2b9a6098d02">python37Packages.python-engineio_3</a></li></li>
    <li><a href="https://gist.github.com/a800b01e44f431072b859dba46738294">python37Packages.python-socketio_4</a></li></li>
    <li><a href="https://gist.github.com/6b73d48a4b1c787ee26a9bf1fd4a70f8">python38Packages.python-engineio_3</a></li></li>
    <li><a href="https://gist.github.com/82fec094431488778a6a0eb110c5bf9b">python38Packages.python-socketio_4</a></li></li>
    <li><a href="https://gist.github.com/a8bb1442fd6c46cb37b48be31a332a16">python39Packages.python-engineio_3</a></li></li>
    <li><a href="https://gist.github.com/cabb6859841179ea2b119baca5512129">python39Packages.python-socketio_4</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `six`.
Near pkgs/development/python-modules/python-engineio/3.nix:10:3:
```
   |
10 | , six
   |   ^
```
</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113247` at 6763803a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/bbb36e6c0e44975d13664b4b21518694">python37Packages.python-engineio_3</a></li></li>
    <li><a href="https://gist.github.com/976b0d076e53460497c5d6233bacbb08">python37Packages.python-socketio_4</a></li></li>
    <li><a href="https://gist.github.com/d381655e5467e183b0dec14db7e41453">python38Packages.python-engineio_3</a></li></li>
    <li><a href="https://gist.github.com/8660222cc3d0609ab98376727ea9c4c5">python38Packages.python-socketio_4</a></li></li>
    <li><a href="https://gist.github.com/ac64ba7d98c4c37bc982c31b9dab1add">python39Packages.python-engineio_3</a></li></li>
    <li><a href="https://gist.github.com/abe4d88d78479b338b8b66a4ed98f21b">python39Packages.python-socketio_4</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `six`.
Near pkgs/development/python-modules/python-engineio/3.nix:10:3:
```
   |
10 | , six
   |   ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -3499,6 +3499,12 @@
     githubId = 6893840;
     name = "Yacine Hmito";
   };
+  graham33 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3502]: Please put this into a separate commit with the message ``maintainers: add graham33``.

File: pkgs/development/python-modules/python-engineio/3.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{ lib, stdenv
+, buildPythonPackage
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-engineio/3.nix:2]:
```suggestion
{ lib
, stdenv
, buildPythonPackage
```

File: pkgs/development/python-modules/python-engineio/3.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    eventlet
+    mock
+    requests
+    tornado
+    websocket_client
+    websockets
+    pytestCheckHook
+  ];
+
+  doCheck = !stdenv.isDarwin;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-engineio/3.nix:39]: Why do we not want to run this on darwin?

File: pkgs/development/python-modules/python-engineio/3.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  preCheck = lib.optionalString stdenv.isLinux ''
+    echo "nameserver 127.0.0.1" > resolv.conf
+    export NIX_REDIRECTS=/etc/protocols=${iana-etc}/etc/protocols:/etc/resolv.conf=$(realpath resolv.conf) \
+      LD_PRELOAD=${libredirect}/lib/libredirect.so
+  '';
+  postCheck = "unset NIX_REDIRECTS LD_PRELOAD";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-engineio/3.nix:46]:
```suggestion
  postCheck = ''
    unset NIX_REDIRECTS LD_PRELOAD
  '';
```

File: pkgs/development/python-modules/python-socketio/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/python-socketio/default.nix:39]: Please fix the commit message for that change.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/python-engineio/3.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
+{ lib, stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, aiohttp
+, eventlet
+, iana-etc
+, libredirect
+, mock
+, requests
+, six
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-engineio/3.nix:10]:
```suggestion
```

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (graham33) [reply maintainers/maintainer-list.nix:3502]: It is a separate commit already, but I've fixed the message.

File: pkgs/development/python-modules/python-engineio/3.nix (hunk 5 above)
Comment (graham33) [reply pkgs/development/python-modules/python-engineio/3.nix:39]: This was copied from pkgs/development/python-modules/python-engineio/default.nix, I don't know if it works or not but it seems like checks are enabled for engineio 3.x in 20.09, so I'll remove this.

File: pkgs/development/python-modules/python-engineio/3.nix (hunk 7 above)
Comment (graham33) [reply pkgs/development/python-modules/python-engineio/3.nix:10]: Actually this should be a propagatedBuildInput, fixed, thanks.

File: pkgs/development/python-modules/python-engineio/3.nix (hunk 6 above)
Comment (graham33) [reply pkgs/development/python-modules/python-engineio/3.nix:48]: Again copied from default.nix/previous version in 20.09.  I'll fix though.

File: pkgs/development/python-modules/python-engineio/3.nix (hunk 4 above)
Comment (graham33) [reply pkgs/development/python-modules/python-engineio/3.nix:2]: Similar in default.nix, but I'll fix.
